clap_complete/src/dynamic/completer.rs

```diff
@@ -53,6 +53,7 @@ pub fn complete(
     let mut current_cmd = &*cmd;
     let mut pos_index = 1;
     let mut is_escaped = false;
+    let mut _state = ParseState::Unknown;
```

Comment:
Do we need an `Unknown` state vs `ValueDone`

Reply:
I think that we need an `Unknown` state. `ValueDone` means we recognize that the word is a subcommand, flag or positional. `Unknown` means we don't know what is meaning of the word, for example, `--flag_unknon`, we parse a word starting with `--` but not a arg belong to the `current_cmd`

Reply:
The first thing suspicious to me is that clap's parser doesn't need it, see https://github.com/clap-rs/clap/blob/5da658c5a06e369a5dc4c073f3f043cab05c403a/clap_builder/src/parser/parser.rs#L1605-L1610

Its also not clear what value we are supposed to be getting out of `Unknown`.     clap's `ValuesDone` is more of the default state of the state machine.  When you aren't currently in a specialized option or positional state, you move back to that.   It sounds like `Unknown` is your default state, but then how is that different than your `ValueDone`?

Granted, it might be clearer once the state is in use.  We should only include in this PR what gets used.

Reply:
`Unknown` and `ValueDone` are indeed both default states of the state machine for now, I will remove Unknown first.